File: Northwind_Demo/.origin/Sources/NorthWind.Web.Angular/src/app/domain/serverActionServices/codeBehind/codeBehindReportService.js
export default class CodeBehindReportService{
	constructor(
		//--  custom dependencies
		//-- /custom dependencies
		customActionRunner) {
		this.customActionRunner = customActionRunner;
		this.serviceRoute = 'reportservice';
	}
		serverPrintSimple (){
			const path = `${this.serviceRoute}/serverprintsimple`;
			return this.customActionRunner.runReportAction(path);
		}
		ServerPrintSimpleCanExecute (){
		    return true;
		}
		ServerPrintSimpleGetConfirmMessage (){
		    return null;
		}
		serverPrintWithParameter ( id){
			const path = `${this.serviceRoute}/serverprintwithparameter?id=${encodeURI(id)}`;
			return this.customActionRunner.runReportAction(path);
		}
		ServerPrintWithParameterCanExecute (){
		    return true;
		}
		ServerPrintWithParameterGetConfirmMessage (){
		    return null;
		}
		serverPrintWithForm ( employeeId,  fromDate,  toDate,  useExcel,  customerId){
			const path = `${this.serviceRoute}/serverprintwithform?employeeId=${encodeURI(employeeId)}&fromDate=${encodeURI(fromDate)}&toDate=${encodeURI(toDate)}&useExcel=${encodeURI(useExcel)}&customerId=${encodeURI(customerId)}`;
			return this.customActionRunner.runReportAction(path);
		}
		ServerPrintWithFormCanExecute (){
		    return true;
		}
		ServerPrintWithFormGetConfirmMessage (){
		    return null;
		}
}

File: Northwind_Demo/.origin/Sources/NorthWind.Client.ServerServices/ActionServices/Server/CodeBehind/CodeBehindReportService.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Net.Http;
using BusinessFramework.Client.Contracts.Connection;
using BusinessFramework.Client.Contracts.Files;
using BusinessFramework.Contracts.Actions;
using BusinessFramework.Contracts.Reporting;
using NorthWind.Client.ServerServices.Properties;
using NorthWind.Client.Services.Contracts.ActionServices;
using NorthWind.Contracts;


namespace NorthWind.Client.ServerServices.ActionServices.Server.CodeBehind
{

    /// <summary>
    /// Code behind
    /// </summary>
    public abstract class CodeBehindReportService : IReportService
    {    

        /// <summary>
        /// Ctor
        /// </summary>
        protected CodeBehindReportService(IControllerClientFactory clientFactory)
        {
            ControllerClient = clientFactory.Create("reportservice");
        }

        /// <summary>
        ///  Server controller client
        /// </summary>
        protected IControllerClient ControllerClient { get; private set; }

		/// <inheritdoc />
		public virtual ActionResult<Report> ServerPrintSimple()
		{
		    var uri = ControllerClient.CreateUri().AddPart("serverprintsimple");
		
		
		    return ControllerClient.HttpPost<ActionResult<Report>>(uri.ToString());
		}

		/// <inheritdoc />
		public virtual bool ServerPrintSimpleCanExecute()
		{
		    return true;
		}
		/// <inheritdoc />
		public virtual string ServerPrintSimpleGetConfirmMessage()
		{
		    return ActionServiceResources.ReportService_ServerPrintSimple_ConfirmationMessage;
		}
		/// <inheritdoc />
		public virtual string ServerPrintSimpleGetWaitMessage()
		{
		    return ActionServiceResources.ReportService_ServerPrintSimple_WaitMessage;
		}
		/// <inheritdoc />
		public virtual ActionResult<Report> ServerPrintWithParameter(int id)
		{
		    var uri = ControllerClient.CreateUri().AddPart("serverprintwithparameter")
		    .AddParameter("id", id);
		
		
		    return ControllerClient.HttpPost<ActionResult<Report>>(uri.ToString());
		}

		/// <inheritdoc />
		public virtual bool ServerPrintWithParameterCanExecute()
		{
		    return true;
		}
		/// <inheritdoc />
		public virtual string ServerPrintWithParameterGetConfirmMessage()
		{
		    return ActionServiceResources.ReportService_ServerPrintWithParameter_ConfirmationMessage;
		}
		/// <inheritdoc />
		public virtual string ServerPrintWithParameterGetWaitMessage()
		{
		    return ActionServiceResources.ReportService_ServerPrintWithParameter_WaitMessage;
		}
		/// <inheritdoc />
		public virtual ActionResult<Report> ServerPrintWithForm(int? employeeId, DateTimeOffset? fromDate, DateTimeOffset? toDate, bool useExcel, int? customerId)
		{
		    var uri = ControllerClient.CreateUri().AddPart("serverprintwithform")
		    .AddParameter("employeeId", employeeId)
		    .AddParameter("fromDate", fromDate)
		    .AddParameter("toDate", toDate)
		    .AddParameter("useExcel", useExcel)
		    .AddParameter("customerId", customerId);
		
		
		    return ControllerClient.HttpPost<ActionResult<Report>>(uri.ToString());
		}

		/// <inheritdoc />
		public virtual bool ServerPrintWithFormCanExecute()
		{
		    return true;
		}
		/// <inheritdoc />
		public virtual string ServerPrintWithFormGetConfirmMessage()
		{
		    return ActionServiceResources.ReportService_ServerPrintWithForm_ConfirmationMessage;
		}
		/// <inheritdoc />
		public virtual string ServerPrintWithFormGetWaitMessage()
		{
		    return ActionServiceResources.ReportService_ServerPrintWithForm_WaitMessage;
		}
    }
}